### PR TITLE
[MM-43785] Insights feature intro

### PR DESCRIPTION
**Pull request by BenCookie95**
#### Summary
Added the feature intro in order to give the feature some visibility. 
Screen Shot 2022-05-17 at 4 18 56 PM

If you dismiss or click the `View insights` button you will no longer get the intro.

- New users who just signed up should not get this modal along with users still in the following flows:
  - First sysadmins who are going through the onboarding flow
  - People who still have the steps modal in the screen below:
Screen Shot 2022-05-16 at 12 54 11 PM
  - People who still have the steps below
Screen Shot 2022-05-13 at 11 03 19 AM

- If you are an existing user, you will get this modal to inform you of the new feature

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-43785

#### Related Pull Requests

- https://github.com/mattermost/mattermost-server/pull/20214

#### Release Note

```release-note
NONE
```

### CODE REVIEW

**Review by mkraft — APPROVED** [review]

**Review by plant99 — APPROVED** [review]

**Comment by matthewbirtch** [comment]
@furqanmlk could you enable the feature flag?

**Comment by furqanmlk** [comment]
> Access here: https://mattermost-webapp-pr-10268.test.mattermost.cloud
@matthewbirtch 
I have added the feature flag for this server in Split.io but it is not getting enabled. 
I sent message to Nick to investigate. I will update here as soon as I get solution.

**Comment by furqanmlk** [comment]
@matthewbirtch Server has Feature Flag is enabled now 

<p dir="auto" style="box-sizing: border-box; margin-top: 0px; margin-bottom: 16px; color: rgb(36, 41, 47); font-family: -apple-system, &quot;system-ui&quot;, &quot;Segoe UI&quot;, Helvetica, Arial, sans-serif, &quot;Apple Color Emoji&quot;, &quot;Segoe UI Emoji&quot;; font-size: 14px; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; white-space: normal; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; background-color: rgb(255, 255, 255); text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;">Access here:<span> </span><a href="https://mattermost-webapp-pr-10268.test.mattermost.cloud/" rel="nofollow" style="box-sizing: border-box; background-color: transparent; color: var(--color-accent-fg); text-decoration: none; transition: color 80ms cubic-bezier(0.33, 1, 0.68, 1) 0s, background-color, box-shadow, border-color;">https://mattermost-webapp-pr-10268.test.mattermost.cloud</a></p>

Account Type | Username | Password
-- | -- | --
Admin | sysadmin | Sys@dmin123
User | user-1 | User-1@123

Access here: https://mattermost-webapp-pr-10268.test.mattermost.cloud/

Account Type	Username	Password
Admin	sysadmin	Sys@dmin123
User	user-1	User-1@123

**Review by matthewbirtch — CHANGES_REQUESTED** [review]
@BenCookie95 I'm seeing some weirdness on first view of this:

1. The arrow isn't pointing to the pulsing dot.
2. The arrow on the tour point is not positioned correctly in relation to the whole tour point container (compared to the design)
3. The arrow is showing strangely (see below). I'm not sure what's causing this though.
image
4. The sidebar 'Insights' item should be highlighted (or knocked out of the darkening layer)

Here is a comparison between implementation and design
<table>
<tr>
<td>Implemented</td>
<td>Designed</td>
</tr>
<tr>
<td>
image
</td>
<td>
image
</td>
</tr>
</table>

**Comment by BenCookie95** [comment]
> @BenCookie95 I'm seeing some weirdness on first view of this:
> 
> 1. The arrow isn't pointing to the pulsing dot.
> 2. The arrow on the tour point is not positioned correctly in relation to the whole tour point container (compared to the design)
> 3. The arrow is showing strangely (see below). I'm not sure what's causing this though.
> 
> image
> 
> 4. The sidebar 'Insights' item should be highlighted (or knocked out of the darkening layer)
> Here is a comparison between implementation and design
> 
> Implemented	Designed
> image	image

Thanks for reviewing, I hadn't seen the positioning issue when threads is enabled. I'll take a look!

**Comment by matthewbirtch** [comment]
One other thing I'm thinking about - we should only show this to existing users. I don't want this to show for users who are new and go through the onboarding.

**Comment by furqanmlk** [comment]
@BenCookie95 can you please fix lint issues ?

**Comment by BenCookie95** [comment]
@furqanmlk I still have one more issue to take care of, will let you know

**Comment by BenCookie95** [comment]
@matthewbirtch I have updated the description, hopefully I have covered all the issues you listed

**Comment by matthewbirtch** [comment]
Looks like I need the feature flag turned on again for this since we had to spin up a new test server @furqanmlk

**Comment by BenCookie95** [comment]
> Looks like I need the feature flag turned on again for this since we had to spin up a new test server @furqanmlk

@matthewbirtch @furqanmlk The cloud test server will have to be created on the server PR in order to pick up the server changes

File: components/activity_and_insights/activity_and_insights_link/insights_tour_tip/insights_tour_tip.tsx
Comment (pvev) [inline components/activity_and_insights/activity_and_insights_link/insights_tour_tip/insights_tour_tip.tsx:9]: Please, group the imports by context and separate those groups by blank lines.

File: components/activity_and_insights/activity_and_insights_link/insights_tour_tip/insights_tour_tip.tsx
Comment (pvev) [inline components/activity_and_insights/activity_and_insights_link/insights_tour_tip/insights_tour_tip.tsx:53]: same here
```suggestion
        id='activityAndInsights.tutorial_tip.viewInsights'
```

**Comment by pvev** [comment]
This ticket https://mattermost.atlassian.net/browse/MM-42023 has the details about the changes made to display the New Onboarding Task list to all users (firstAdmin, Subsequent admins and end users) and also, all the code related to the legacy `Recommended Next Steps` is removed. This is the PR https://github.com/mattermost/mattermost-webapp/pull/10187.  So there is no need to cover the Recommended Next Steps scenario in these changes.

File: components/activity_and_insights/activity_and_insights_link/insights_tour_tip/insights_tour_tip.tsx
Comment (neallred) [inline components/activity_and_insights/activity_and_insights_link/insights_tour_tip/insights_tour_tip.tsx:87]: @pvev as another outcome of the [show the onboarding task list to all users PR] (https://github.com/mattermost/mattermost-webapp/pull/10187), the `(firstAdmin && showTaskList)` condition will need to be changed to `showTaskList` once that PR is is merged, right?

I think the whole `!(useCaseOnboarding && firstAdmin)` condition can be removed. If use case onboarding is false, `showTaskList` will not be true, because it is code in the onboarding that sets the state showTaskList is based on.

**Review by neallred — APPROVED** [review]
Approving as I don't think my comments are blocking, but the `firstAdmin` condition should be removed once @pvev 's PR is merged.

File: components/activity_and_insights/activity_and_insights_link/insights_tour_tip/insights_tour_tip.tsx
Comment (pvev) [reply components/activity_and_insights/activity_and_insights_link/insights_tour_tip/insights_tour_tip.tsx:87]: @neallred is right. The condition must change cause all the user will have the onboarding task list.

**Comment by furqanmlk** [comment]
> Looks like I need the feature flag turned on again for this since we had to spin up a new test server @furqanmlk

@matthewbirtch Done

File: components/activity_and_insights/activity_and_insights_link/insights_tour_tip/insights_tour_tip.tsx
Comment (BenCookie95) [reply components/activity_and_insights/activity_and_insights_link/insights_tour_tip/insights_tour_tip.tsx]: I've updated this to match your new PR I think @pvev. Let me know if the `if statement` is correct

**Comment by matthewbirtch** [comment]
@BenCookie95 I'm not quite sure how to get the tour point to show up. Is there something I need to do. I tried creating a new account, but can't get it to show.

File: components/activity_and_insights/activity_and_insights_link/insights_tour_tip/insights_tour_tip.tsx
Comment (BenCookie95) [reply components/activity_and_insights/activity_and_insights_link/insights_tour_tip/insights_tour_tip.tsx]: Ok that's great! Thanks

**Comment by BenCookie95** [comment]
> @BenCookie95 I'm not quite sure how to get the tour point to show up. Is there something I need to do. I tried creating a new account, but can't get it to show.

@matthewbirtch I've made it visible for the sysadmin now. Let me know if it disappears and need it back, I had to do a cURL request to enable it as the preference is automatically hidden for new users

**Review by matthewbirtch — APPROVED** [review]
LGTM!

**Review by pvev — APPROVED** [review]
LGTM! thanks @BenCookie95 . The PR for removing the nextStep deprecated feature is getting a final review and will be merged soon. I will let you know as soon as it gets merged. Sorry for the delay on this. Thanks!

File: components/activity_and_insights/activity_and_insights_link/insights_tour_tip/insights_tour_tip.tsx
Comment (pvev) [inline components/activity_and_insights/activity_and_insights_link/insights_tour_tip/insights_tour_tip.tsx:64]: all the validations related to nextSteps are not longer relevant. Also, the nextSteps was removed from the state.

Comment (BenCookie95) [reply components/activity_and_insights/activity_and_insights_link/insights_tour_tip/insights_tour_tip.tsx]: The check onboarding function only contains one thing  now:
```
const isOnboardingOngoing = useCallback(() => {
        if (showTaskList) {
            return true;
        }
        return false;
}, [showTaskList]);
```

**Comment by BenCookie95** [comment]
> @BenCookie95 all looks good except few unexpected behaviour
> 
> image

These are fixed now

**Comment by furqanmlk** [comment]
@BenCookie95  There is no feature intro in Cloud instance
https://mattermost-webapp-pr-10268.test.mattermost.cloud/ 
Admin: user-mattermost-webapp-pr-10268@example.mattermost.com
Password: Cws@10102020

**Review by furqanmlk — COMMENTED** [review]
@BenCookie95  There is no feature intro in Cloud instance
https://mattermost-webapp-pr-10268.test.mattermost.cloud/
Admin: [user-mattermost-webapp-pr-10268@example.mattermost.com](mailto:user-mattermost-webapp-pr-10268@example.mattermost.com)
Password: Cws@10102020

**Comment by BenCookie95** [comment]
> user-mattermost-webapp-pr-10268@example.mattermost.com

@furqanmlk just re-enable the intro by sending a curl request like this:
```
curl 'https://mattermost-webapp-pr-10268.test.mattermost.cloud/api/v4/users/kz34wgzo4j8a8kiuekzosmf3ph/preferences' \
  -X 'PUT' \
  -H 'Accept: */*' \
  -H 'Accept-Language: en' \
  -H 'Connection: keep-alive' \
  -H 'Content-Type: application/json' \
  -H 'Cookie: MMAUTHTOKEN=6ktntkxarf8dubefrrcs4tiryc; MMUSERID=kz34wgzo4j8a8kiuekzosmf3ph; MMCSRF=saw4dmym4tgodkkqqtnyb5bo8o; MMCLOUDURL=mattermost-webapp-pr-10268' \
  -H 'Sec-Fetch-Dest: empty' \
  -H 'Sec-Fetch-Mode: cors' \
  -H 'Sec-Fetch-Site: same-origin' \
  -H 'User-Agent: Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/101.0.4951.64 Safari/537.36' \
  -H 'X-CSRF-Token: saw4dmym4tgodkkqqtnyb5bo8o' \
  -H 'X-Requested-With: XMLHttpRequest' \
  -H 'sec-ch-ua: " Not A;Brand";v="99", "Chromium";v="101", "Google Chrome";v="101"' \
  -H 'sec-ch-ua-mobile: ?0' \
  -H 'sec-ch-ua-platform: "macOS"' \
  --data-raw '[{"user_id":"kz34wgzo4j8a8kiuekzosmf3ph","category":"insights","name":"insights_tutorial_state","value":"{\"insights_modal_viewed\":false}"}]' \
  --compressed
```

You also still had the onboarding steps visible so that stopped it from appearing. It should appear now

**Review by furqanmlk — APPROVED** [review]
Working as expected

**Review by furqanmlk — APPROVED** [review]

**Pull request merged**
